buffer/src/main/java/io/netty/buffer/AbstractReferenceCountedDuplicatedByteBuf.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+import io.netty.util.IllegalReferenceCountException;
+import io.netty.util.internal.PlatformDependent;
+
+public abstract class AbstractReferenceCountedDuplicatedByteBuf extends DuplicatedAbstractByteBuf {
```

Comment:
Why we need this class at all and it can not be part of the `PooledDuplicatedByteBuf` ? Seems like `PooledDuplicatedByteBuf` is the only that extend it.